src/chainlit/server.py

```diff
@@ -410,36 +407,38 @@ async def connection_successful(sid):
     if config.code.lc_factory:
         """Instantiate the langchain agent and store it in the session."""
         agent = await config.code.lc_factory()
-        session["agent"] = agent
+        session.agent = agent
 
     if config.code.llama_index_factory:
         llama_instance = await config.code.llama_index_factory()
-        session["llama_instance"] = llama_instance
+        session.llama_instance = llama_instance
 
 
 @socket.on("disconnect")
 async def disconnect(sid):
-    if sid in sessions:
-        # Clean up the session
-        sessions.pop(sid)
+    async def disconnect_on_timeout(sid):
+        await asyncio.sleep(config.project.session_timeout)
+        if session := Session.get(sid):
+            # Clean up the user session
+            if session.id in user_sessions:
+                user_sessions.pop(session.id)
+            # Clean up the session
+            session.delete()
```

Comment:
Don't we want to check if the session was active recently before deleting it? It looks more like a scheduled deletion rather than a timeout here, is it what we want?

Reply:
Should not be issue as we get the session by `socket_id` before deleting it, if a session has been restored (meaning that it is still used), the `socket_id` is replaced and the session will not be found after the timeout. The deletion will be handled when the new socket will be disconnected.